src/clr-angular/forms/common/providers/ng-control.service.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable} from "@angular/core";
+import {NgControl} from "@angular/forms";
+import {Observable, Subject} from "rxjs";
+
+@Injectable()
+export class NgControlService {
+    // Observable to subscribe to the control, since its not available immediately for projected content
+    private _controlChanges: Subject<NgControl> = new Subject<NgControl>();
+    get controlChanges(): Observable<NgControl> {
+        return this._controlChanges.asObservable();
+    }
+
+    private _control: NgControl;
+    get control(): NgControl {
+        return this._control;
+    }
+
+    set control(control: NgControl) {
+        this._control = control;
+        this._controlChanges.next(control);
```

Comment:
Why do we need a setter and a getter when the `Subject` emits the `NgControl`?

Could we not have just used a method which registers the `NgControl` and all its subscribers know about it because the `Subject` emits the control on registration?